### ISSUE TITLE
Delivery to mailu mailboxes is sometimes delayed, sometimes fail silently

### ISSUE THREAD

**Issue by sourcegr** [Mailu/Mailu#2863]
### Environment

- docker compose

### Version

- Version: `2.0`

## Description
Delivery to `mailu` mailboxes is sometimes delayed, sometimes fail silently. 

## Replication Steps   
- does not apply -

## Observed behaviour
With `mailu` instwlled (`docker-compose`), delivery to local mailboxes is sometimes delayed and sometimes not delivered at all - it fails silently. For example, sometimes when I send an email from a gmail account to a `mailu` email account on the server, the email is not delivered. I cannot see any connections to the mailu server  from google servers via the `docker-compose logs --follow`. Gmail does not report any problems, and the email seems to be sent, but it is not delivered to the `mailu` mailbox.

There are few times that the email is delivered very fast, other times that the  email is delivered with a significant delay, and sometimes that the email is not delivered at all. We use the `mailjet` service to send automated emails to an email account, and `mailjet` seems unable to deliver the message, thus marking the email as `retrying` and after a day it soft-bounces and gets lost!

## Expected behaviour
I would expect the email to be delivered in a reasonable time.

## Logs

I get a lot of errors, but I believe all of them are irrelevant to the pain point - although the will be fixed in the near future. any help with  these errors would be very appreciated.
Containers not listed bellow, naaamely `antivirus / admin / webmail / front / resolver / oletools / redis` produce no errors

```
docker-compose logs --tail=200 imap | grep -i error

imap_1       | Jun 21 03:22:10 lmtp(3745): Error: SSL context initialization failed, disabling SSL: Can't load SSL certificate (ssl_cert setting): The certificate is empty
```

```
docker-compose logs --tail=200 front | grep -i err

front_1      | 2023/06/21 05:08:56 [crit] 2389#2389: *2797 SSL_shutdown() failed (SSL: error:0A000123:SSL routines::application data after close notify) while connecting to upstream, client: 87.253.233.103 using starttls, server: 0.0.0.0:25
```

```
docker-compose logs --tail=200 antispam | grep -i err

antispam_1   | 2023-06-21 02:20:43 #157(rspamd_proxy) <00796e>; lua; rbl.lua:248: error looking up XXX.XXX.XXX.XXX.bl.blocklist.de: server fail
antispam_1   | 2023-06-21 02:25:11 #157(rspamd_proxy) <7d4ba6>; spf; spf_record_dns_callback: spf error for domain a1565682.bnc3.mailjet.com: cannot resolve AAAA record for a1565682.bnc3.mailjet.com: requested record is not found
(many of the above with different IPs and hostnames form mailjet)
```

My `docker-compose` is
```
version: '2.2'
services:
  # External dependencies
  redis:
    image: redis:alpine
    restart: always
    volumes:
      - "/mailu/redis:/data"
    depends_on:
      - resolver
    dns:
      - 192.168.203.254

  # Core services
  front:
    image: ${DOCKER_ORG:-ghcr.io/mailu}/${DOCKER_PREFIX:-}nginx:${MAILU_VERSION:-2.0}
    restart: always
    env_file: mailu.env
    logging:
      driver: journald
      options:
        tag: mailu-front
    ports:
      - "127.0.0.1:9080:80"
      - "127.0.0.1:9443:443"
      - "148.251.78.236:25:25"
      - "148.251.78.236:465:465"
      - "148.251.78.236:587:587"
      - "148.251.78.236:110:110"
      - "148.251.78.236:995:995"
      - "148.251.78.236:143:143"
      - "148.251.78.236:993:993"
    networks:
      - default
      - webmail
    volumes:
      - "/mailu/certs:/certs"
      - "/mailu/overrides/nginx:/overrides:ro"
    depends_on:
      - resolver
    dns:
      - 192.168.203.254

  resolver:
    image: ${DOCKER_ORG:-ghcr.io/mailu}/${DOCKER_PREFIX:-}unbound:${MAILU_VERSION:-2.0}
    env_file: mailu.env
    restart: always
    networks:
      default:
        ipv4_address: 192.168.203.254

  admin:
    image: ${DOCKER_ORG:-ghcr.io/mailu}/${DOCKER_PREFIX:-}admin:${MAILU_VERSION:-2.0}
    restart: always
    env_file: mailu.env
    logging:
      driver: journald
      options:
        tag: mailu-admin
    volumes:
      - "/mailu/data:/data"
      - "/mailu/dkim:/dkim"
    depends_on:
      - redis
      - resolver
    dns:
      - 192.168.203.254

  imap:
    image: ${DOCKER_ORG:-ghcr.io/mailu}/${DOCKER_PREFIX:-}dovecot:${MAILU_VERSION:-2.0}
    restart: always
    env_file: mailu.env
    logging:
      driver: journald
      options:
        tag: mailu-imap
    volumes:
      - "/mailu/mail:/mail"
      - "/mailu/overrides/dovecot:/overrides:ro"
    depends_on:
      - front
      - resolver
    dns:
      - 192.168.203.254

  smtp:
    image: ${DOCKER_ORG:-ghcr.io/mailu}/${DOCKER_PREFIX:-}postfix:${MAILU_VERSION:-2.0}
    restart: always
    env_file: mailu.env
    logging:
      driver: journald
      options:
        tag: mailu-smtp
    volumes:
      - "/mailu/mailqueue:/queue"
      - "/mailu/overrides/postfix:/overrides:ro"
    depends_on:
      - front
      - resolver
    dns:
      - 192.168.203.254

  oletools:
    image: ${DOCKER_ORG:-ghcr.io/mailu}/${DOCKER_PREFIX:-}oletools:${MAILU_VERSION:-2.0}
    hostname: oletools
    restart: always
    networks:
      - noinet
    depends_on:
      - resolver
    dns:
      - 192.168.203.254

  antispam:
    image: ${DOCKER_ORG:-ghcr.io/mailu}/${DOCKER_PREFIX:-}rspamd:${MAILU_VERSION:-2.0}
    hostname: antispam
    restart: always
    env_file: mailu.env
    logging:
      driver: journald
      options:
        tag: mailu-antispam
    networks:
      - default
      - noinet
    volumes:
      - "/mailu/filter:/var/lib/rspamd"
      - "/mailu/overrides/rspamd:/overrides:ro"
    depends_on:
      - front
      - redis
      - oletools
      - antivirus
      - resolver
    dns:
      - 192.168.203.254

  # Optional services
  antivirus:
    image: ${DOCKER_ORG:-ghcr.io/mailu}/${DOCKER_PREFIX:-}clamav:${MAILU_VERSION:-2.0}
    restart: always
    env_file: mailu.env
    volumes:
      - "/mailu/filter:/data"
    depends_on:
      - resolver
    dns:
      - 192.168.203.254



  # Webmail
  webmail:
    image: ${DOCKER_ORG:-ghcr.io/mailu}/${DOCKER_PREFIX:-}webmail:${MAILU_VERSION:-2.0}
    restart: always
    env_file: mailu.env
    volumes:
      - "/mailu/webmail:/data"
      - "/mailu/overrides/snappymail:/overrides:ro"
    networks:
      - webmail
    depends_on:
      - front

networks:
  default:
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: 192.168.203.0/24
  webmail:
    driver: bridge
  noinet:
    driver: bridge
    internal: true
```

and the `mailu.env`
```
MAILU_VERSION=2.0.14
SECRET_KEY=xxxxxxxxxxxxxxxx
SUBNET=192.168.203.0/24
DOMAIN=my-domain-name.gr
HOSTNAMES=my-domain-name.gr,mail.my-domain-name.gr,power.my-domain-name.gr
POSTMASTER=admin
TLS_FLAVOR=cert
AUTH_RATELIMIT_IP=5000/hour
AUTH_RATELIMIT_USER=5000/day
DISABLE_STATISTICS=False
ADMIN=true
WEBMAIL=snappymail
API=false
WEBDAV=none
ANTIVIRUS=none
SCAN_MACROS=false
MESSAGE_SIZE_LIMIT=50000000
MESSAGE_RATELIMIT=20000/day
RELAYNETS=
RELAYHOST=
FETCHMAIL_ENABLED=False
FETCHMAIL_DELAY=600
RECIPIENT_DELIMITER=+
DMARC_RUA=admin
DMARC_RUF=admin
WELCOME=false
WELCOME_SUBJECT=Welcome to your new email account
WELCxOME_BODY=Welcome to your new email account, if you can read this, then it is configured properly!
COMPRESSION=
COMPRESSION_LEVEL=
WEBROOT_REDIRECT=/webmail
WEB_ADMIN=/admin
WEB_WEBMAIL=/webmail
WEB_API=
SITENAME=my-domain-name.gr
WEBSITE=https://my-domain-name.gr
COMPOSE_PROJECT_NAME=mailu
CREDENTIAL_ROUNDS=12
REJECT_UNLISTED_RECIPIENT=
LOG_LEVEL=WARNING
TZ=Europe/Athens
DEFAULT_SPAM_THRESHOLD=80
API_TOKEN=
```

Feel free to ask for anything more,

Thanks in advance,

**Comment by sourcegr** [comment]
Some extra details

the error is "Internal Server Error" like in this example

```
root@machine /mailu/mail # telnet <M.Y.I.P.> 25
Trying M.Y.I.P....
Connected to M.Y.I.P.
Escape character is '^]'.
220 <host> ESMTP ready
EHLO <host>
250-<host>
250-PIPELINING
250-SIZE 50000000
250-ETRN
250-ENHANCEDSTATUSCODES
250-8BITMIME
250-DSN
250 STARTTLS
MAIL FROM:<random@email.com>
250 2.0.0 OK
RCPT TO:<randomagain@email.com>
451 4.3.2 Internal server error
Connection closed by foreign host.
```

**Comment by nextgens** [comment]
Please share the matching log lines from "front". It should give additional information.

**Comment by sourcegr** [comment]
@nextgens thank you very much, but we really had enough trying and failing, so we decided to ditch the whole project and use some other tools; we are not looking for a solution anymore - I will close this issue right after this post.

It's been four horrible days trying to set this up correctly, and I am sorry to say that we are kind of happy we decided to go replace it.

It strikes me, though, that - after looking at all the issues here - I found that a lot of people have the same or almost the same issue and nobody actually solved it - unless they solved it and did not disclose the solution.
